src/ssl.mli

```diff
@@ -209,9 +209,16 @@ type protocol =
   | SSLv23 (** accept all possible protocols (SSLv2 if supported by openssl,
                SSLv3, TLSv1, TLSv1.1, TLSv1.2, and TLSv1.3) *)
   | SSLv3 (** only SSL v3 protocol *)
-  | TLSv1 (** only TLS v1 protocol *)
-  | TLSv1_1 (** only TLS v1.1 protocol *)
+  | TLSv1
+    [@ocaml.alert deprecated
```

Comment:
I think this is still good to include and merge. but not the `-Wno-deprecated-declarations` call.

Reply:
I like the fact that I don't have 5 pages of C warnings possibly hiding more important stuff. We could have the Gcc pragma then. I don't think it will break compilation under other compiler, but will remove the unwanted warning ?

Reply:
I just remove the compilation flag, It is indeed useless with openSSL 1.1.1